ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx

```diff
@@ -38,31 +45,448 @@ export function showErrorMessages(errors: any) {
+  message: '',
+  defaultValue: '',
+}
+
+function is2DArray(coordinates: string) {
+  try {
+    const parsedCoords = JSON.parse(coordinates)
+    return Array.isArray(parsedCoords) && Array.isArray(parsedCoords[0])
+  } catch (e) {
+    return false
+  }
+}
+
+function hasPointError(point: any[]) {
+  if (
+    point.length !== 2 ||
+    Number.isNaN(Number.parseFloat(point[0])) ||
+    Number.isNaN(Number.parseFloat(point[1]))
+  ) {
+    return true
+  }
+  return point[0] > 180 || point[0] < -180 || point[1] > 90 || point[1] < -90
+}
+
+function getGeometryErrors(filter: any): Set<string> {
+  const geometry = filter.geojson && filter.geojson.geometry
+  const errors = new Set<string>()
+  if (!geometry) {
+    return errors
+  }
+  const properties = filter.geojson.properties
+  const bufferWidth = properties.buffer && properties.buffer.width
+  switch (properties.type) {
+    case 'Polygon':
+      if (!Array.isArray(geometry.coordinates[0]) || !geometry.coordinates[0].length) {
+        errors.add(
+          'Polygon coordinates must be in the form [[x,y],[x,y],[x,y],[x,y], ... ]'
+        )
+      } else if (geometry.coordinates[0].length < 4) {
+        // check for MultiPolygon
+        geometry.coordinates[0].forEach((shape: number[]) => {
+          if (shape.length < 4) {
+            errors.add(
+              'Polygon coordinates must be in the form [[x,y],[x,y],[x,y],[x,y], ... ]'
+            )
+          }
+        })
+      }
+      break
+    case 'LineString':
+      if (!Array.isArray(geometry.coordinates) || !geometry.coordinates.length || geometry.coordinates.length < 2) {
+        errors.add('Line coordinates must be in the form [[x,y],[x,y], ... ]')
+      }
+      // Can't just check !bufferWidth because of the case of the string "0"
+      if (bufferWidth === undefined || Number(bufferWidth) <= 0) {
+        errors.add('Line buffer width must be greater than 0')
+      }
+      break
+    case 'Point':
+      if (bufferWidth === undefined || Number(bufferWidth) <= 0) {
+        errors.add('Radius must be greater than 0')
+      }
+      if (
+        geometry.coordinates.some(
+          (coord: any) => !coord || coord.toString().length === 0
+        )
+      ) {
+        errors.add('Coordinates must not be empty')
+      }
+      break
+    case 'BoundingBox':
+      const { east, west, north, south } = filter.geojson.properties
+      if (
+        [east, west, north, south].some(
+          direction => direction === '' || direction === undefined
+        )
+      ) {
+        errors.add('Bounding box must have valid values')
+      }
+      break
   }
   return errors
 }
+
+function validateLinePolygon(mode: string, currentValue: string) {
+  if (!is2DArray(currentValue)) {
+    return { error: true, message: 'Not an acceptable value' }
+  }
+  try {
+    return validateListOfPoints(JSON.parse(currentValue), mode)
+  } catch (e) {
+    return { error: true, message: 'Not an acceptable value' }
+  }
+}
```

Comment:
✏️ If you changed this function to this:
`function validateLinePolygon(mode: string, currentValue: string) {`
  `try {`
    `const parsedCoords = JSON.parse(currentValue)`
    `if (!is2DArray(parsedCoords)) {`
      `return { error: true, message: 'Not an acceptable value' }`
    `}`
    `return validateListOfPoints(parsedCoords, mode)`
  `} catch (e) {`
    `return { error: true, message: 'Not an acceptable value' }`
  `}`
`}`

then `is2DArray` could be simplified to:
`function is2DArray(coordinates: string) {`
  `return Array.isArray(parsedCoords) && Array.isArray(parsedCoords[0])`
`}`